recipe/2024-04-terraform-provider-secrets.md

```diff
@@ -0,0 +1,259 @@
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Terraform Provider | A Terraform provider is a plugin that allows Terraform to interact with and manage resources of a specific infrastructure platform or service, such as AWS, Azure, or Google Cloud. |
+
+## Objectives
+
+**Reference for new type recipeConfig and handling of secrets for Private Terraform Repository:** [Design document for Private Terraform Repository](https://github.com/radius-project/design-notes/blob/3644b754152edc97e641f537b20cf3d87a386c43/recipe/2024-01-support-private-terraform-repository.md)
+
+**Secret Implementation for Private Terraform Repository:**[Secret Implementation for Private Terraform Repository](https://github.com/radius-project/radius/pull/7306)
+
+**Reference for support of multiple Terraform Providers in Radius:** [Design Document to support multiple Terraform Providers](https://github.com/radius-project/design-notes/pull/39/files)
+
+> **Issue Reference:** <!-- (If appropriate) Reference an existing issue that describes the feature or bug. -->
+https://github.com/radius-project/radius/issues/6539
+
+### Goals
+
+Describe data flow, specific function calls when using secrets to configure Terraform Providers, building on design in the referred design document.
+
+### Non goals
+
+The document is focussed on handling secrets for Terraform Providers, with Applications.Core/secretStores being the underlying resource for the secret store (which today supports Kubernetes secrets). Other sources and types of secretsStores are out of scope for this document. 
+
+### User scenarios (optional)
+
+#### User story 1
+As an operator, I maintain a set of Terraform recipes for use with Radius. I have a set of provider configurations which are applicable across multiple recipes. These provider configurations now include secrets, which are injected into the Terraform configuration, allowing the Terraform process to access them as needed.
+
+
+## User Experience (if applicable)
+
+**Sample Input:**
+
+This is pulled in from existing document:
+[Design Document to support multiple Terraform Providers](https://github.com/radius-project/design-notes/pull/39/files)
+
+``` diff
+resource env 'Applications.Core/environments@2023-10-01-preview' = {
+  name: 'dsrp-resources-env-recipes-context-env'
+  location: 'global'
+  properties: {
+    compute: {
+      ...  
+    }
+    providers: {
+      ...
+    }
+    recipeConfig: {
+      terraform: {
+        authentication: {
+          ...
+        }
+        providers: {
+          azurerm: [
+            {
+              subscriptionid: 1234
+              tenant_id: '745fg88bf-86f1-41af'
++              secrets: { // Individual Secrets from SecretStore
```

Comment:
Are there any `rad CLI` commands such as `rad recipe show` or `rad env show` that need to display any information to users when a secret is registered? Such as letting a user know that secrets are configured or is that something users would have to find through the Bicep file definition?

Reply:
Per discussion, this is something PMs may want to think about ahead.

Reply:
@lakshmimsft was there an action item that came out of this?

Reply:
tagging @jasonviviano for any further thoughts

Reply:
I'd like PMs to create an user story around a CLI experience for this as a separate work item.

Reply:
I'm not convinced that we need a CLI experience for this, so I vote we wait for user feedback to dictate whether we should build it.